terraform/environments/prd/aws/main.tf

```diff
@@ -48,13 +48,44 @@ module "alb" {
   sg_alb_description   = each.value.sg_alb_description
   sg_alb_rules_ingress = each.value.sg_alb_rules_ingress
   sg_alb_rules_egress  = each.value.sg_alb_rules_egress
-  #  sg_alb_create_before_destroy = local.sg_alb_create_before_destroy
   # TGs
   tgs_alb    = each.value.tgs_alb
   tg_lb_type = each.value.tg_lb_type
-  # listeners
-  alb_listener_80  = each.value.alb_listener_80
+  # listener http
+  alb_listener_80 = each.value.alb_listener_80
+}
+
+module "alb_listener_https" {
+  # ALB HTTPS submodule
+  source   = "../../../modules/aws/alb/alb_listener_https"
+  for_each = local.albs
+  # from ALB
+  alb_arn = module.alb[each.value.vpc_name].lb_arn
+  alb_tg  = module.alb[each.value.vpc_name].lb_tg
+  # from route53
+  alb_route53_certificate_arn = module.route53[each.value.vpc_name].certificate_arn
+  # listener https
   alb_listener_443 = each.value.alb_listener_443
-  # listeners rules
+  # listener https rules
   alb_rules = each.value.alb_rules
 }
+
+module "route53" {
+  source   = "../../../modules/aws/route53/"
+  for_each = local.routes53
+  # alb outputs
+  route53_alb_dns_name = module.alb[each.value.vpc_name].dns_name
+  route53_alb_zone_id  = module.alb[each.value.vpc_name].zone_id
+  # zone
+  zone_name = each.value.dns_name
+  # A_record
+  a_record_name   = each.value.record_name
```

Comment:
Hmm what if would need to create multiple records here of different type? :thinking:

Reply:
the thing is that the modules interface is a bit confusing. So we pass `route53_alb_dns_name` and `route53_alb_zone_id` and `a_record_*` vars and then used them in single record resource

Reply:
Can we pass to the module  a list or map  of alias records and  list  or map of `normal` records and base on that create the records we need?

Reply:
1. multiple records - yes, different type - no (or need write more code and use dynamic blocks). 2. I renamed variables. 3. we take alias from another resource, we do not enter the manual here. therefore, we cannot pass them, there is no point in it

Reply:
I'll think about how to do it simply and for a large number of records